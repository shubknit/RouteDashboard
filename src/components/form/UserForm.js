import React, { Component } from 'react';

export class UserForm extends Component {
    constructor(){
        super();
        this.state = {
            buttonText: 'Submit'
        };
        this._startPoint = {
            value: ''
        };
        this._endPoint = {
            value: ''
        };
    }
    submit = (e) => {
        e.preventDefault();
        this.setState({
            buttonText: 'Re-Submit'
        })
        this.props.addLocation({
            start : this._startPoint.value,
            end : this._endPoint.value
        })
    }

    render(){
        return (
            <div className = 'form-container'>
                <form onSubmit = {this.submit} className = 'form'>
                    <label htmlFor = 'start-point'> Starting Location </label>
                    <input type ="text" id ="start-point" placeholder ='start point' className = 'text-input' 
                    ref = { input => this._startPoint = input  } required/>
    
                    <label htmlFor = 'end-point'> Drop-off point</label>
                    <input type ="text" id ="end-point" placeholder ='end point' className = 'text-input'
                    ref = { input => this._endPoint = input } required/>
                    {this.props.data.isFetching && <p>Loading...</p>}
                    {this.props.data.mapData.distance && <p> total distance: {this.props.data.mapData.distance} </p>}
                    {this.props.data.mapData.time && <p> total time: {this.props.data.mapData.time} </p>}
                    {this.props.data.error && <p className = 'error'>  {this.props.data.error} </p>}
                    <div className = 'button-container'><button  className = 'button' type ="submit"> {this.state.buttonText} </button>
                        <button className = 'button' type ="reset"> Reset </button>
                     </div>
                    </form>
            </div>
        )
    }
   
}