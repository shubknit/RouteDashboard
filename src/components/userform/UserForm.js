import React, { Component } from 'react';
import { ViewRouteDetails } from '../routedetail/ViewRouteDetails';
import { InputFieldAutoComplete } from  '../../shared/components/InputFieldAutoComplete';

export class UserForm extends Component {
    constructor(){
        super();
        this.state = {
            buttonText: 'Submit'
        };
    }
    
    // update states(startingLocation and DropOffPoint) on selecting the places from autocomplete
    handleOnSelectAddress = (locationDetail, name) => {
        this.setState({
            [name]: locationDetail,
        })
    }

    // Below method enables or disables the submit button
    enableSubmit = () => {
        const { startingLocation, dropOffPoint } = this.state;
         return startingLocation && dropOffPoint;
    }

    // handles when user click submit button
    submit = (e) => {
        const { startingLocation, dropOffPoint } = this.state;
        e.preventDefault();
        this.setState({
            buttonText: 'Re-Submit',
           
        })
        this.props.addLocation({
            start : startingLocation,
            end : dropOffPoint
        })
    }

    // handles when user click reset button
    resetForm = () => {
        this.setState({
            buttonText: 'Submit'
        })
        this.props.resetForm();
    }

    render(){
        return (
            <div className = 'form-container'>
                <form onSubmit = {this.submit} className = 'form'>
                    <InputFieldAutoComplete 
                        label="Starting location"
                        handleOnSelectAddress={this.handleOnSelectAddress}
                        inputName="startingLocation"
                    />
                    <InputFieldAutoComplete 
                        label="Drop-off point"
                        handleOnSelectAddress={this.handleOnSelectAddress}
                        inputName="dropOffPoint"
                    />
                    <ViewRouteDetails data = { this.props.data }/>
                    <div className = 'button-container'>
                    <button  className = 'button' type ="submit" disabled={!this.enableSubmit()}> {this.state.buttonText} </button>
                    <button className = 'button' type ="reset" onClick = {this.resetForm}> Reset </button>
                     </div>
                </form>
            </div>
        )
    }
   
}