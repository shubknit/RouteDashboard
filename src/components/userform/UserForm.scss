.button {
    background: $blue;
    border: none;
    color: white;
    display: inline-block;
    font-size: $font-size-large;
    font-weight: 300;
    line-height: 1;
    padding: $s-size;
    text-decoration: none;
}

.button--link {
    background: none;
}

.button--secondary {
    background: #888;
}

.text-input {
    border: 1px solid #cacccd;
    height: 50px;
    font-size: $font-size-large;
    font-weight: 300;
    padding: $s-size;
}

.select {
    @extend .text-input;
}

.textarea {
    @extend .text-input;
    height: 10rem;
}

.form {
    display: flex;
    flex-direction: column;

    >* {
        margin-bottom: $m-size;
    }
    label{
        display: block;
    }
}

.form__error {
    margin: 0 0 $m-size 0;
    font-style: italic;
}
.error {
    color: red;
}