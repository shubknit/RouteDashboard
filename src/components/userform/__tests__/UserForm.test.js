import React from 'react';
import { UserForm } from '../UserForm';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

const data = {
    mapData: {
        distance: '200',
        time: 20
    }
}
const errorData = {
    mapData: {
      
    },
    error: 'location not accesbile'
}
let startLocation;
let dropOffPoint;



it('should render the form correctly with distance and time', () => {
    const wrapper = renderer.create(<UserForm data = {data}/>).toJSON();
    expect(wrapper).toMatchSnapshot();
})

it('should render the error correctly', () => {
    const wrapper = renderer.create(<UserForm data = {errorData}/>).toJSON();
    expect(wrapper).toMatchSnapshot();
})

it('should submit form correctly', () => {
    const onSubmitSpy = jest.fn();
	const wrapper = shallow(<UserForm data = {data} addLocation = {onSubmitSpy}/>)
    wrapper.find('form').simulate('submit', {
    	preventDefault: () => {}
    })
	expect(onSubmitSpy).toHaveBeenLastCalledWith({
        start : startLocation,
        end : dropOffPoint
	})
})