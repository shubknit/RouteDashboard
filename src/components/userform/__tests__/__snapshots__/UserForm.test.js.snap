// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the error correctly 1`] = `
<div
  className="form-container"
>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <label
      htmlFor="start-point"
    >
       Starting Location 
    </label>
    <input
      className="text-input"
      id="start-point"
      placeholder="start point"
      required={true}
      type="text"
    />
    <label
      htmlFor="end-point"
    >
       Drop-off point
    </label>
    <input
      className="text-input"
      id="end-point"
      placeholder="end point"
      required={true}
      type="text"
    />
    <div
      className="route-info"
    >
      <p
        className="error"
      >
          
        location not accesbile
         
      </p>
    </div>
    <div
      className="button-container"
    >
      <button
        className="button"
        type="submit"
      >
         
        Submit
         
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="reset"
      >
         Reset 
      </button>
    </div>
  </form>
</div>
`;

exports[`should render the form correctly with distance and time 1`] = `
<div
  className="form-container"
>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <label
      htmlFor="start-point"
    >
       Starting Location 
    </label>
    <input
      className="text-input"
      id="start-point"
      placeholder="start point"
      required={true}
      type="text"
    />
    <label
      htmlFor="end-point"
    >
       Drop-off point
    </label>
    <input
      className="text-input"
      id="end-point"
      placeholder="end point"
      required={true}
      type="text"
    />
    <div
      className="route-info"
    >
      <p>
         total distance: 
        200
         
      </p>
      <p>
         total time: 
        20
         
      </p>
    </div>
    <div
      className="button-container"
    >
      <button
        className="button"
        type="submit"
      >
         
        Submit
         
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="reset"
      >
         Reset 
      </button>
    </div>
  </form>
</div>
`;
